File: Event-management/Event-Booking-System/src/main/java/com/bluepal/service/PaymentService.java
package com.bluepal.service;

import com.bluepal.dto.PaymentRequest;
import com.bluepal.model.Payment;
import com.bluepal.response.RazorPayOrderResponse;
import com.razorpay.RazorpayException;
import java.util.List;

public interface PaymentService {

    // Create RazorPay order and save Payment in MongoDB
    RazorPayOrderResponse createOrder(PaymentRequest paymentRequest) throws RazorpayException;

    // Verify RazorPay payment signature and update MongoDB; also confirm booking on success
    Payment verifyPayment(String razorpayOrderId, String razorpayPaymentId, String razorpaySignature);

    // Get Payment by bookingId from MongoDB
    Payment getPaymentByBookingId(String bookingId);

    // Update Payment status in MongoDB
    Payment updatePaymentStatus(String paymentId, String status);

    // List all payments (admin use)
    List<Payment> getAllPayments();

    // Verify a payment by scanning (uses Razorpay API to fetch and validate)
    Payment scanVerify(String razorpayOrderId, String razorpayPaymentId);
}

File: Event-management/Event-Booking-System/my-project/src/src/services/paymentService.js
import api from './api';

// Create a new payment order
export const createOrder = async (data) => {
  try {
    const response = await api.post('/payments/create-order', data);
    return response.data;
  } catch (error) {
    console.error('Error creating order:', error);
    throw error;
  }
};

// Verify payment after completion
export const verifyPayment = async (orderId, paymentId, signature) => {
  try {
    const response = await api.post('/payments/verify', null, {
      params: {
        razorpayOrderId: orderId,
        razorpayPaymentId: paymentId,
        razorpaySignature: signature,
      },
    });
    return response.data;
  } catch (error) {
    console.error('Error verifying payment:', error);
    throw error;
  }
};

// Get payment details by booking ID
export const getPaymentByBookingId = async (bookingId) => {
  try {
    const response = await api.get(`/payments/booking/${bookingId}`);
    return response.data;
  } catch (error) {
    console.error('Error fetching payment:', error);
    throw error;
  }
};

// Get all payments (admin)
export const getAllPayments = async () => {
  try {
    const response = await api.get('/payments');
    return response.data;
  } catch (error) {
    console.error('Error fetching all payments:', error);
    throw error;
  }
};
